Neos.Media/Classes/Domain/Service/AssetService.php

```diff
@@ -284,28 +290,37 @@ public function replaceAssetResource(AssetInterface $asset, PersistentResource $
             /** @var AssetVariantInterface $variant */
             foreach ($variants as $variant) {
                 $originalVariantResource = $variant->getResource();
-                try {
-                    $presetIdentifier = $variant->getPresetIdentifier();
-                    $variantName = $variant->getPresetVariantName();
-                    $variant = $this->assetVariantGenerator->recreateVariant($asset, $presetIdentifier, $variantName);
-                    if ($variant === null) {
-                        $this->logger->debug(
-                            sprintf('No variant returned when recreating asset variant %s::%s for %s', $presetIdentifier, $variantName, $asset->getTitle()),
+                $presetIdentifier = $variant->getPresetIdentifier();
```

Comment:
As you use just methods of the ImageVariant Model maybe use ImageVariant instead of AssetVariantInterface.
Or extend the AssetVariantInterface if the methods should be available also for other Variant Models.

Reply:
The code is intended for "generic use", but for now we don't have asset variants "that much". The whole code around asset variants is a bit "limited" but planned "extensible". And it's not the `ImageVariantService`, but the `AssetVariantService`. So, I'd like to leave it like that. And ajdusting the interface has to wait until 6.0, sadly.